http/src/main/scala/org/http4s/blaze/http/http2/StreamManager.scala

```diff
@@ -1,11 +1,82 @@
 package org.http4s.blaze.http.http2
 
-// Place holder
+import org.http4s.blaze.http._
+
+import scala.concurrent.Future
+
+/** Manager of the active streams for a session
+  *
+  * The `StreamManager` can be thought of as the collection of
+  * active streams and some associated helper methods for performing
+  * operations relevant to the HTTP/2 protocol.
+  */
 private trait StreamManager {
 
+  /** The `StreamIdManager` owned by this `StreamManager` */
+  def idManager: StreamIdManager
+
+  /** Number of active streams */
   def size: Int
 
-  def streamFinished(stream: StreamState, cause: Option[Http2Exception]): Unit
+  /** Whether any streams are active */
+  def isEmpty: Boolean
+
+  /** Update the flow windows of open streams due to a change of the initial flow window
+    *
+    * A receiver MUST adjust the size of all stream flow-control windows that
+    * it maintains by the difference between the new value and the old value.
+    * https://tools.ietf.org/html/rfc7540#section-6.9.2
+    *
+    * @param delta difference between the new initial window and the previous initial window
+    */
+  def initialFlowWindowChange(delta: Int): MaybeError
+
+  /** Get the stream associated with the specified stream ID */
+  def get(streamId: Int): Option[StreamState]
+
+  /** Register an `InboundStreamState` with the set of active streams */
+  def registerInboundStream(state: InboundStreamState): Boolean
 
+  /** Register an `OutboundStreamState` with the set of active streams
+    *
+    * @return the newly allocated stream id to be associated with the new
+    *         outbound stream, it it was possible to allocate one.
+    */
   def registerOutboundStream(state: OutboundStreamState): Option[Int]
+
+  /** Cause the associated stream to be reset, if it exists
+    *
+    * @param cause the reason the stream was reset
+    * @return true if the stream existed and was closed, false otherwise
+    */
+  def rstStream(cause: Http2StreamException): Boolean
+
+  /** Called by a `StreamState` to remove itself from the StreamManager
+    *
+    * @param streamState the stream being closed
+    * @return true if the stream existed and was closed, false otherwise
+    */
+  def streamClosed(streamState: StreamState): Boolean
```

Comment:
Instead of `streamFinished` we now have `streamClosed` which returns a `Boolean` to signal whether the stream was still considered active by the session. This puts the onus on `StreamState` to manage its own exceptions appropriately, and it can escalate session level exceptions if needed.